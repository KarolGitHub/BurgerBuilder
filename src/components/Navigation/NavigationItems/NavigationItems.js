import React from "react";

import NavigationItem from "./NavigationItem/NavigationItem";
import classes from "./NavigationItems.module.css";

const navigationItems = (props) => (
  <ul className={classes.NavigationItems}>
    <NavigationItem link="/" exact>
      Burger Builder
    </NavigationItem>
    {props.isAuth ? (
      <NavigationItem link="/orders">Orders</NavigationItem>
    ) : null}
    {props.isAuth ? (
      <NavigationItem link="/logout">Log Out</NavigationItem>
    ) : (
      <NavigationItem link="/login">Log In</NavigationItem>
    )}
  </ul>
);

export default navigationItems;
