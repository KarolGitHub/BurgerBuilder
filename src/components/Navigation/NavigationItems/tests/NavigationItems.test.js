import React from 'react';
import { shallow } from 'enzyme';

import NavigationItems from '../NavigationItems';
import NavigationItem from '../NavigationItem/NavigationItem';


const testComponent = <NavigationItems />;

describe('<NavigationItems />', () => {

    describe('Unit tests', () => {
		let wrapper;
		beforeEach(() => {
			wrapper = shallow(testComponent);
		});

		it('should render two <NavigationItem/> elements if not authenticated', () => {
			expect(wrapper.find(NavigationItem)).toHaveLength(2);
		});

		it('should render three <NavigationItem/> elements if authenticated', () => {
			wrapper.setProps({ isAuth: true });
			expect(wrapper.find(NavigationItem)).toHaveLength(3);

		});

		it('should render three <NavigationItem/> elements if authenticated', () => {
			wrapper.setProps({ isAuth: true });
			expect(wrapper.contains(<NavigationItem link='/logout'>Log Out</NavigationItem>)).toEqual(true);

		});

		it('snapshot match', () => {
			expect(wrapper).toMatchSnapshot();
		});	
		
	});
});

