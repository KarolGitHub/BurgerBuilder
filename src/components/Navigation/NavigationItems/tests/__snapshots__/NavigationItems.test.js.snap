// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItems /> Render unit tests snapshot match 1`] = `ShallowWrapper {}`;

exports[`<NavigationItems /> Unit tests snapshot match 1`] = `ShallowWrapper {}`;
